unit_tests/src/test_get_delay_rule_info.cpp

```diff
@@ -0,0 +1,108 @@
+#include "catch.hpp"
+
+#include "client_connection.hpp"
+#include "getRodsEnv.h"
+#include "get_delay_rule_info.h"
+#include "irods_at_scope_exit.hpp"
+#include "irods_query.hpp"
+#include "rodsClient.h"
+#include "rodsDef.h"
+#include "rodsErrorTable.h"
+#include "user_administration.hpp"
+
+#include <json.hpp>
+
+#include <cstdlib>
+#include <string>
+#include <utility>
+
+TEST_CASE("rc_get_delay_rule_info")
+{
+    using json = nlohmann::json;
+
+    load_client_api_plugins();
+
+    irods::experimental::client_connection conn;
+    auto* conn_ptr = static_cast<RcComm*>(conn);
+
+    SECTION("delay rule exists")
+    {
+        // Schedule a delay rule and get its ID.
+        REQUIRE(std::system(R"_(irule -r irods_rule_engine_plugin-irods_rule_language-instance 'delay("<INST_NAME>irods_rule_engine_plugin-irods_rule_language-instance</INST_NAME>") { writeLine("serverLog", "Testing API plugin."); }' null ruleExecOut)_") == 0);
+
+        std::string rule_id;
+        for (auto&& row : irods::query{conn_ptr, "select RULE_EXEC_ID"}) {
+            rule_id = row[0];
+            break;
+        }
+
+        INFO("GenQuery returned rule_id = " << rule_id);
+
+        BytesBuf* bbuf{};
+        REQUIRE(rc_get_delay_rule_info(conn_ptr, rule_id.c_str(), &bbuf) == 0);
+
+        CHECK(bbuf->buf);
+        CHECK(bbuf->len > 0);
+
+        const auto* p = static_cast<char*>(bbuf->buf);
+        const auto info = json::parse(p, p + bbuf->len);
+
+        CHECK(info.size() == 13);
+        CHECK(info.at("rule_id").get_ref<const std::string&>() == rule_id);
+        CHECK(info.count("rule_name"));
+        CHECK(info.count("rei_file_path"));
+        CHECK(info.count("user_name"));
+        CHECK(info.count("exe_address"));
+        CHECK(info.count("exe_time"));
+        CHECK(info.count("exe_frequency"));
+        CHECK(info.count("priority"));
+        CHECK(info.count("last_exe_time"));
+        CHECK(info.count("exe_status"));
+        CHECK(info.count("estimated_exe_time"));
+        CHECK(info.count("notification_addr"));
+        CHECK(info.count("exe_context"));
+    }
+
+    SECTION("invalid input")
+    {
+        BytesBuf* bbuf{};
+        CHECK(rc_get_delay_rule_info(conn_ptr,     nullptr,   &bbuf) == SYS_INVALID_INPUT_PARAM);
+        CHECK(rc_get_delay_rule_info(conn_ptr,     "00000", nullptr) == SYS_INVALID_INPUT_PARAM);
+        CHECK(rc_get_delay_rule_info(conn_ptr,        "-1",   &bbuf) ==       CAT_NO_ROWS_FOUND);
+        CHECK(rc_get_delay_rule_info(conn_ptr, "bad_input",   &bbuf) ==       SYS_LIBRARY_ERROR);
```

Comment:
Is an empty string (`""`) considered valid input? What happens?

Reply:
Done.